.buildkite/scripts/publish/publish-common.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+if [[ "${CURDIR:-}" == "" ]]; then
+  echo "!! CURDIR is not set. Exiting."
+  exit 2
+fi
+
+function realpath {
+  echo "$(cd "$(dirname "$1")"; pwd)"/"$(basename "$1")";
+}
+
+export SCRIPT_DIR="$CURDIR"
+export BUILDKITE_DIR=$(realpath "$(dirname "$SCRIPT_DIR")")
+export PROJECT_ROOT=$(realpath "$(dirname "$BUILDKITE_DIR")")
+
+VERSION_PATH="$PROJECT_ROOT/product_version"
+export VERSION=$(cat $VERSION_PATH)
+
+if [[ "${USE_SNAPSHOT:-}" == "true" ]]; then
+  echo "Adding SNAPSHOT labeling"
+  export VERSION="${VERSION}-SNAPSHOT"
+fi
+
+export BASE_TAG_NAME=${DOCKER_IMAGE_NAME:-docker.elastic.co/enterprise-search/crawler}
```

Comment:
There's been a thread going with leads about what the right repository/namespace is for our stuff going forward. I've bumped the thread, but worth pinging Diana/Dana too. But I don't think `/enterprise-search` is going to be the right place for this, and we might as well start in the right place 🤞

Reply:
Once a decision is made about the future namespace, do you have an idea of how long it would take to set up? Just thinking about how long this PR may have to wait for a decision + implementation.

Reply:
I don't think setup is required - I think you'll just be able to push wherever you tell it to go. But we'll find out!
I'm hoping we can hear back from leads by tomorrow/Monday

Reply:
related to this, there's also a slack thread from a few weeks ago https://elastic.slack.com/archives/C7UQBTBD0/p1722424618694589

and a related github issue: https://github.com/elastic/search-team/issues/7481

Reply:
FYI I've addressed all of the review comments, and I think docker repo namespace is the only blocker now for this PR.

Reply:
Latest I've seen from Paul is to go with `docker.elastic.co/integrations/crawler`. Let's go with that. Worst case scenario, we publish a few images there then have to pivot later.

Reply:
> FYI I've addressed all of the review comments, and I think docker repo namespace is the only blocker now for this PR.

thanks for the fixes, this is looking great to me 🎉 

I'll leave it to you all to decide whether we'd want to consider switching to Wolfi for the base image as a blocker (main rationale being: let's not start by publishing a new image with hundreds of CVEs). I'd guess it depends on how soon we need an official crawler image available.

Reply:
update to the above, it's looking to me like a wolfi-based image may be easily within reach: https://github.com/elastic/crawler/pull/106#issuecomment-2307878798

I was able to build a wolfi-based image locally and start the crawler:
```
$ docker run -it crawler-wolfi:latest
Commands:
  crawler crawl CRAWL_CONFIG                   # Run a crawl of the site
  crawler validate CRAWL_CONFIG                # Validate crawler configuration
  crawler version                              # Print version
```

Reply:
> Latest I've seen from Paul is to go with docker.elastic.co/integrations/crawler. Let's go with that. Worst case scenario, we publish a few images there then have to pivot later.

@seanstory I've updated the base docker tag to be this value.

Reply:
@acrewdson thanks heaps for iterating on the wolfi image. I'd like to have it from day one, but I'd prefer to merge it in from the other PR. Merging this PR won't produce an image yet so I think it's okay to do them separately.

Reply:
> I'd like to have it from day one, but I'd prefer to merge it in from the other PR. Merging this PR won't produce an image yet so I think it's okay to do them separately.

makes sense 👍

Reply:
> Suggestion: could search-integrations be more future-proof than integrations? Please discard this comment if this was already extensively discussed.

@oli-g this was discussed among the leads, but I don't have a thread available to reference. @seanstory is the thread where this was discussed shareable? If not, do you think the current naming is fine?

Reply:
Confirmed that `integrations` is the official answer.